lightwood/encoders/time_series/rnn.py

```diff
@@ -5,122 +5,171 @@
+                        next_tensor, decoder_hidden = self._decoder.forward(
+                            tensor_target[:, tensor_i, :].unsqueeze(dim=1),
+                            decoder_hidden)
+                    else:
+                        next_tensor, decoder_hidden = self._decoder.forward(next_tensor.detach(), decoder_hidden)
+
+                    loss += self.criterion(next_tensor, tensor_target[:, tensor_i + 1, :].unsqueeze(dim=1))
 
-                loss = loss
                 average_loss += int(loss)
                 loss.backward()
-                optimizer.step()
+                self.optimizer.step()
 
-            average_loss = average_loss/len(priming_data)
+            average_loss = average_loss / len(priming_data)
 
             if average_loss < self._stop_on_error:
                 break
             if feedback_hoop_function is not None:
-                feedback_hoop_function("epoch [{epoch_n}/{total}] average_loss = {average_loss}".format(average_loss=average_loss, epoch_n=i+1, total=self._train_iters))
+                feedback_hoop_function("epoch [{epoch_n}/{total}] average_loss = {average_loss}".format(
+                    epoch_n=i + 1,
+                    total=self._train_iters,
+                    average_loss=round(average_loss, 2)))
 
         self._prepared = True
 
-    def _encode_one(self, data, initial_hidden = None, return_next_value = False):
+    def _encode_one(self, data, initial_hidden=None, return_next_value=False):
         """
         This method encodes one single row of serial data
-        :param data: a string representing a list of values separate by space, for example: `1 2 3 4` or a list [1, 2, 3, 4]
+        :param data: multidimensional time series as list of lists [[dim1_data], [dim2_data], ...]
+                     (dim_data: string with format "x11, x12, ... x1n")
         :param initial_hidden: if you want to encode from an initial hidden state other than 0s
-        :param return_next_value:  if you to return the next value in the time series too
+        :param return_next_value:  if you want to return the next value in the time series too
 
-        :return:  either encoded_value or encoded_value, next_value
+        :return:  either encoded_value or (encoded_value, next_value)
         """
         self._encoder.eval()
         with torch.no_grad():
-
-            data_tensor = tensor_from_series(data, self.device)
-
+            data_tensor = tensor_from_series(data, self.device, n_dims=self._n_dims,
+                                             pad_value=self._eos, max_len=self._max_ts_length)
+            steps = data_tensor.shape[1]
             encoder_hidden = self._encoder.initHidden(self.device)
             encoder_hidden = encoder_hidden if initial_hidden is None else initial_hidden
-            if type(encoder_hidden) is list:
-                encoder_hidden = torch.Tensor([[encoder_hidden]], device=device)
+
             next_tensor = None
```

Comment:
Unless `steps` can ever be == 0 this needn't be declared here.

Reply:
I am not sure if I understand, can you elaborate?